registry/api/v2/errors.go

```diff
@@ -24,7 +24,7 @@ var (
 		Description: `The access controller denied access for the operation on
 		a resource. Often this will be accompanied by a 401 Unauthorized
 		response status.`,
-		HTTPStatusCode: http.StatusForbidden,
+		HTTPStatusCode: http.StatusUnauthorized,
```

Comment:
In my quick scan of the docker code it never looks for a 403 ( http.statusforbidden ) but does look for 401 ( http.statusunauthorized ) in several spots.